raiden/tests/unit/transfer/mediated_transfer/test_mediatorstate.py

```diff
@@ -2112,67 +2112,25 @@ def test_imbalance_penalty_prevents_transfer():
     1. Only transfer as much as he has capacity. If this payment succeeds, this
        is to the mediator's advantage, since he gets to keep more tokens.
     2. Refund the transfer, because he can't send the negative imbalance fee he
-    promised and the payment is likely to have enough tokens when reaching the
+    promised and the payment is unlikely to have enough tokens when reaching the
     target.
     This test verifies that we choose option 2.
     """
-    # Will reward payment with the enormous amount of 1 token per transferred token!
-    imbalance_penalty = [(0, 100), (100, 0)]
+    # Will reward payment with the amount of 1 token per 10 transferred token!
+    imbalance_penalty = [(0, 0), (1000, 100)]
     pair, _ = _foward_transfer_pair(
         10,
         NettingChannelStateProperties(
             our_state=NettingChannelEndStateProperties(balance=10),
-            fee_schedule=FeeScheduleState(flat=0, imbalance_penalty=imbalance_penalty),
+            fee_schedule=FeeScheduleState(cap_fees=False),
         ),
         NettingChannelStateProperties(
             our_state=NettingChannelEndStateProperties(balance=10),
-            fee_schedule=FeeScheduleState(flat=0, imbalance_penalty=imbalance_penalty),
-        ),
-    )
-    assert not pair
-
-
-def test_outdated_imbalance_penalty_at_transfer():
```

Comment:
We update the FeeScheduleState whenever the balance changes. So the case that is exercised by this test should be impossible.

Reply:
What about testing this for a regression?

Reply:
That would be a different test. This test checks that we don't raise an exception when we get into an invalid state. The proper regression test would be to check that we don't get into an invalid state in the first place.
I don't know what that test would look like, yet. It would somehow include an onchain balance change and a mediation, I guess.

Reply:
all right understood